renderdoc/driver/gl/gl_initstate.cpp

```diff
@@ -380,6 +391,7 @@ bool GLResourceManager::Prepare_InitialState(GLResource res, byte *blob)
   {
     VAOInitialData *data = (VAOInitialData *)blob;
 
+    RDCASSERT(!data->valid);
```

Comment:
@baldurk I haven't had any major problems with these changes, but I did hit this assert once during a capture, after quite a long session with many captures.

The only thing I can think of is that a context got deleted during a capture.  I'll see if i can get more info if it happens again.

Reply:
Even if there might be something still up, these changes look good now so I'm happy to merge them (I've made a couple of stylistic comments).

Regarding this assert firing, I don't have any immediate ideas. If the context was deleted and recreated during a capture, and if we created a new VAO with the same resource name on the 'new' context we shouldn't go back to preparing it. The list of prepared resources is fixed at the start of the frame capture: Some will be prepared immediately, some will be queued. I could see maybe a false negative, where we go to prepare an old object and get a new ID with no blob associated, but not the other way around.

It might be interesting (depending on how often you can repro) to see if it's being prepared twice, or if we're poking into free'd memory or something. I'm hoping it's not the latter but it'd be better to check.

I wonder if this might be due to a design problem in the GL driver. In all other drivers, the API objects are wrapped so you can go from object handle -> ID directly as the wrapper object contains its ID. Thus even if you get two identical object handles there won't be any ambiguity about their ID. In GL, we have the lookup table from resource to ID, so if the same resource comes around twice there might be edge cases there.